Data transposition - state column based

emigre459 opened the pull request:
This code uses the State column of the PPP dataset to identify a large transposition error in which 86% of loans have values in columns that are actually positioned two to the left of the column they actually should be in. This code identifies the problem records, flags them in a new column, then fixes them and flags them as fixed (in another new column).

emigre459 committed:
Update environment.yml to include viz packages and start exploring

emigre459 committed:
Use State column to identify and fix values transposed relative to proper columns

emigre459 committed:
Add custom local package to environment.yml

emigre459 committed:
Improve version control in docker image

emigre459 committed:
Flag problematic and programmatically-fixed rows and columns

emigre459 committed:
Add better handling of problem/fix flag columns when transposing

emigre459 committed:
Add dvc functionality to allow for GDrive data file versioning

emigre459 committed:
Setup dvc with authorization to Google Drive

Remote name is "gdrive"

emigre459 committed:
Output transposition-cleaned CSV and track via dvc

emigre459 committed:
Leverage performance boosts from pandas Categoricals with cleaned data

See an increase of 49x speed for State value_counts() operation

kbmorales commented:
Great work! We should replicate this in the R data ingest pipe as well. Did you make a flat file for the resulting PPP data by chance? I'd love to pair that up with NAICS to see if that helps with the failed matches I found.

emigre459 committed:
Fix JobsRetained mixed values

It was strings and integers, not just integers. Also added
new pandas 1.0 Int64 type to the cleaned column so that it could
support nulls.

kaogilvie committed:
added service account function back in for reference

kaogilvie committed:
adding in service account switch

kaogilvie committed:
added service acct and an optional cleaned csv dump

kaogilvie reviewed (approved):
Got through all of this and it looks GREAT. I especially like the extensible cleaned data section with all those functions. Will come in handy.

I added in the service account auth again, just because it's the most intuitive for me in terms of "how" to do it in Google Console. OAuth InstalledApps led me down the path of needing to get a bunch of approvals from Google, so I balked and reverted to the service user.

emigre459 committed:
Merge in Google service account changes

emigre459 committed:
Merge branch 'master' into data_tranposition
